external/sparqlquery-service.js

```diff
@@ -0,0 +1,244 @@
+                                ?assertionId schema:hasKeywords ?keyword .
+                                ${publicVisibilityQuery}
+                                ${filterQuery}
+                            }
+                        ${limitQuery}`;
+
+        const result = await this.execute(sparqlQuery);
+        return result;
+    }
+
+    async findAssetsByKeyword(query, options, localQuery) {
+        if (options.prefix && !(typeof options.prefix === 'boolean')) {
+            this.logger.error(`Failed FindAssetsByKeyword: ${options.prefix} is not a boolean`);
+            throw new Error('Prefix is not an boolean');
+        }
+        if (localQuery && !(typeof localQuery === 'boolean')) {
+            this.logger.error(`Failed FindAssetsByKeyword: ${localQuery} is not a boolean`);
+            throw new Error('Localquery is not an boolean');
+        }
+        query = this.cleanEscapeCharacter(query);
+        const limitQuery = this.createLimitQuery(options);
+
+        const publicVisibilityQuery = !localQuery ? 'schema:hasVisibility "public" :' : '';
+        const filterQuery = options.prefix ? this.createFilterParameter(query, this.filtertype.KEYWORDPREFIX) : this.createFilterParameter(query, this.filtertype.KEYWORD);
+        const issuerFilter = options.issuers ? this.createFilterParameter(options.issuers, this.filtertype.ISSUERS) : '';
+        const typesFilter = options.types ? this.createFilterParameter(options.types, this.filtertype.TYPES) : '';
+
+        const sparqlQuery = `PREFIX schema: <http://schema.org/>
+                            SELECT ?assertionId
+                            WHERE {
+                                ?assertionId schema:hasTimestamp ?latestTimestamp ;
+                                 ${publicVisibilityQuery}
+                                                     schema:hasUALs ?assetId .
+                                    {
+                                        SELECT ?assetId (MAX(?timestamp) AS ?latestTimestamp)
+                                        WHERE {
+                                            ?assertionId schema:hasKeywords ?keyword ;
+                                                         schema:hasIssuer ?issuer ;
+                                                         schema:hasType ?type ;
+                                                         schema:hasTimestamp ?timestamp ;
+                                                         schema:hasUALs ?assetId .
+                                ${filterQuery}
+                                ${issuerFilter}
+                                ${typesFilter}
+                                        }
+                                        GROUP BY ?assetId
+                                        ${limitQuery}
+                                    }
+                            }`;
+        const result = await this.execute(sparqlQuery);
+        return result;
+    }
+
+    async healthCheck() {
```

Comment:
This probably should not be here, only in triple store service (graphdb/blazegraph) and called from data service

Reply:
Resolved. HealtCheck needs to be implemented by subclass